clang_metashell: fatal error: unknown argument: '-profiler'

I compile metashell in archlinux with `cmake .. -DCMAKE_CXX_FLAGS:STRING="-Werror -DMETASHELL_DISABLE_TEMPLIGHT_TESTS" -DCLANG_STATIC=false -DCLANG_DEBUG=true -DCLANG_LIBRARYDIR=/lib/clang/3.7.0/lib -DNO_TEMPLIGHT=true` like https://travis-ci.org/sabel83/metashell/builds/87318313 .
Then `make metashell` and run it. After
```
#include <vector>
#msh mdb std::vector<int>
```
I get `clang_metashell: fatal error: unknown argument: '-profiler'`.
Both gcc and clang.

I use `./build.sh` to build metashell with gcc.
A link error:
```
...
Scanning dependencies of target metashell_unit_test
[ 69%] Building CXX object test/unit/CMakeFiles/metashell_unit_test.dir/mock_cxtranslationunit.cpp.o
[ 70%] Linking CXX executable metashell
[ 70%] Building CXX object test/unit/CMakeFiles/metashell_unit_test.dir/test_pragma_handler_map.cpp.o
[ 71%] Building CXX object test/unit/CMakeFiles/metashell_unit_test.dir/mock_console.cpp.o
[ 72%] Linking CXX static library libmetashell_system_test_lib.a
/usr/bin/ld: ../../3rd/templight/build/lib/libLLVMSupport.a(Process.cpp.o): undefined reference to symbol 'tigetnum'
/usr/lib/libncursesw.so.6: error adding symbols: DSO missing from command line
collect2: 错误：ld 返回 1
app/CMakeFiles/metashell.dir/build.make:323: recipe for target 'app/metashell' failed
make[2]: *** [app/metashell] Error 1
CMakeFiles/Makefile2:123: recipe for target 'app/CMakeFiles/metashell.dir/all' failed
make[1]: *** [app/CMakeFiles/metashell.dir/all] Error 2
...
```

--- a/build.sh
+++ b/build.sh
@@ -41,7 +41,7 @@ elif [ "$(uname)" = "Darwin" ]
 then
   PLATFORM=osx
   PLATFORM_VERSION=$(sw_vers -productVersion)
-elif [ `cat /etc/lsb-release 2>/dev/null | grep DISTRIB_ID` = "DISTRIB_ID=Ubuntu" ]
+elif [ "`cat /etc/lsb-release 2>/dev/null | grep DISTRIB_ID`" = "DISTRIB_ID=Ubuntu" ]
 then
   PLATFORM=ubuntu
 elif [ -e /etc/debian_version ]

--- a/build.sh
+++ b/build.sh
@@ -71,7 +71,7 @@ echo "Platform: ${PLATFORM}"
 # Build Clang
 cd 3rd
   cd templight
-    mkdir build; cd build
+    mkdir -p build; cd build
       cmake ../llvm -DLIBCLANG_BUILD_STATIC=ON -DCMAKE_BUILD_TYPE=Release \
         && make clang libclang libclang_static templight -j${BUILD_THREADS}
     cd ..
@@ -90,7 +90,7 @@ then
   tools/clang_default_path --gcc=clang > lib/core/extra_sysinclude.hpp
 fi
 
-mkdir bin; cd bin
+mkdir -p bin; cd bin
   cmake .. \
     && make -j${BUILD_THREADS} \
     && make test \

--- a/build.sh
+++ b/build.sh
@@ -72,7 +72,10 @@ echo "Platform: ${PLATFORM}"
 cd 3rd
   cd templight
     mkdir -p build; cd build
-      cmake ../llvm -DLIBCLANG_BUILD_STATIC=ON -DCMAKE_BUILD_TYPE=Release \
+      cmake ../llvm \
+        -DLIBCLANG_BUILD_STATIC=ON \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DLLVM_ENABLE_TERMINFO=OFF \
         && make clang libclang libclang_static templight -j${BUILD_THREADS}
     cd ..
   cd ..

--- a/build.sh
+++ b/build.sh
@@ -47,6 +47,9 @@ then
 elif [ -e /etc/debian_version ]
 then
   PLATFORM=debian
+elif [ -e /etc/arch-release ]
+then
+  PLATFORM=arch
 elif [ `uname` = "FreeBSD" ]
 then
   PLATFORM=freebsd

--- a/build.sh
+++ b/build.sh
@@ -16,6 +16,8 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+set -e
+
 if [ ! -d cmake ]
 then
   echo "Please run this script from the root directory of the Metashell source code"

--- a/docs/getting_metashell/build.md
+++ b/docs/getting_metashell/build.md
@@ -23,7 +23,7 @@
     * `cd 3rd/templight`
     * `mkdir build`
     * `cd build`
-    * `cmake ../llvm -DLIBCLANG_BUILD_STATIC=ON -DCMAKE_BUILD_TYPE=Release`
+    * `cmake ../llvm -DLIBCLANG_BUILD_STATIC=ON -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_TERMINFO=OFF`
     * `make clang libclang libclang_static templight`
 * Now compile Metashell. In the source directory run the following commands:
     * `mkdir bin`

